vim/core/oni-plugin-buffers/index.js

```diff
@@ -0,0 +1,111 @@
+// @ts-check
+const path = require("path")
+
+const activate = Oni => {
+    const menu = Oni.menu.create()
+
+    const truncateFilePath = filepath => {
+        const sections = filepath.split(path.sep)
+        const folderAndFiles = sections.slice(-2)
+        return folderAndFiles.join(path.sep)
+    }
+
+    const updateBufferList = (Oni, menu) => {
+        const buffers = Oni.editors.activeEditor.getBuffers()
+        const active = Oni.editors.activeEditor.activeBuffer.filePath
+
+        const bufferMenuItems = buffers.map(b => ({
+            label: `${active === b.filePath ? b.id + " %" : b.id}`,
+            detail: truncateFilePath(b.filePath),
+            icon: Oni.ui.getIconClassForFile(b.filePath),
+            pinned: active === b.filePath,
+            metadata: { filePath: b.filePath, id: b.id },
+        }))
+
+        return bufferMenuItems
+    }
+
+    const createBufferList = () => {
+        const buffers = updateBufferList(Oni, menu)
+        menu.show()
+        menu.setItems(buffers)
+    }
+
+    const toggleBufferList = () => {
+        !menu.isOpen() ? createBufferList() : menu.hide()
+    }
+
+    const deleteBuffer = async menu => {
+        if (menu.selectedItem) {
+            await Oni.editors.activeEditor.bufferDelete(menu.selectedItem.metadata.id)
+            // This line forces vim to navigate to the next buffer after deleting
+            // NOTE: This is essential as otherwise Oni does not properly register the buffer
+            // delete event
+            await Oni.editors.activeEditor.neovim.command("bn")
```

Comment:
If this will always need to be done along with the `bufferDelete`, perhaps we should just push this inside the `bufferDelete` method?

Reply:
In addition - can we check if the current buffer's id (`activeEditor.activeBuffer.id`) is equal to the one we're deleting? If I understand correctly, we only need to do the `bn` if the buffer is the same, so if `activeEditor.activeBuffer.id === menu.selectedItem.metadata.id`.

Reply:
👍